src/driver/socket.rs

```diff
@@ -44,6 +44,11 @@ impl Socket {
         op.write().await
     }
 
+    pub async fn writev<T: IoBuf>(&self, buf: Vec<T>) -> crate::BufResult<usize, Vec<T>> {
```

Comment:
Could this be generalised to 

```suggestion
    pub async fn writev<T: IoBuf, B: Deref<Target = [T]>>(&self, buf: Buf) -> crate::BufResult<usize, Buf> {
```

Reply:
I'm not really fond of the `Vec<T>` requirement either here or internally:

https://github.com/tokio-rs/tokio-uring/blob/913aa14c567cfae026b7d4f88a5155921cfb9575/src/driver/writev.rs#L22

And I am using `smallvec` in a bunch of places but I'm always wary of stuff moving after it's been sent to the uring submission queue. I guess this small change is probably harmless though? I wonder what @Noah-Kennedy thinks.

Reply:
Ah, we'd also need to make the Future type generic over the buffer type.

Reply:
Making the op futures generic on the buffer type seems to be the direction in general. The logic around op futures was simplified with the Completable trait recently and the Completable trait already is generic on the buffer type. .. If I understand your discussion here correctly ..

Reply:
It seems fine to be following what was done for the File.writev_at at this time.